src/Ext/Techno/Body.cpp

```diff
@@ -552,6 +553,55 @@ void TechnoExt::UpdateMindControlAnim(TechnoClass* pThis)
 	}
 }
 
+
+
+
+static bool CanFireAt(TechnoClass* pTechno, AbstractClass* pTarget)
+{//Is there a already implemented function to tell whether the pTechno can target at pTarget?
+ //Need help here
+	const int wpnIdx = pTechno->SelectWeapon(pTarget);
+	FireError fErr = pTechno->GetFireError(pTarget, wpnIdx, true);
+	if (!(fErr == FireError::ILLEGAL
+		|| fErr == FireError::CANT
+		|| fErr == FireError::MOVING
+		|| fErr == FireError::RANGE))
+	{
+		return pTechno->IsCloseEnough(pTarget, wpnIdx);
+	}
+	else
+		return false;
+}
+void TechnoExt::JumpjetVehicleFacingFix(TechnoClass* pThis)
+{
+	const auto pType = pThis->GetTechnoType();
+	if (pType->JumpJet && pThis->IsInAir() 
+		&& pThis->WhatAmI() == AbstractType::Unit && !pType->Turret)
+	{
+		const auto pFoot = abstract_cast<UnitClass*>(pThis);
+		const auto pTypeExt = TechnoTypeExt::ExtMap.Find(pType);
+		if (pTypeExt && pTypeExt->JumpjetFacing && pFoot->GetCurrentSpeed() == 0)
+		{
+			if (const auto pTarget = pThis->Target)
+			{
+				const auto pLoco = static_cast<JumpjetLocomotionClass*>(pFoot->Locomotor.get());
+				if (pLoco && !pLoco->LocomotionFacing.in_motion() && CanFireAt(pThis, pTarget))
+				{
+					const CoordStruct source = pThis->Location;
+					const CoordStruct target = pTarget->GetCoords();
+					const DirStruct tgtDir = DirStruct(Math::arctanfoo(source.Y - target.Y, target.X - source.X));
+					if (pThis->GetRealFacing().value32() != tgtDir.value32())
+					{
+						pLoco->LocomotionFacing.turn(tgtDir);
+					}
+
+				}
+
+			}
+			
```

Comment:
```suggestion
					if (pThis->GetRealFacing().value32() != tgtDir.value32())
						pLoco->LocomotionFacing.turn(tgtDir);
				}
			}
```
Too much lines, and no need for the braces here